sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -1246,8 +1246,8 @@ std::unique_ptr<CurlNetworkConnection> CurlConnectionPool::ExtractOrCreateCurlCo
     bool resetPool)
 {
   uint16_t port = request.GetUrl().GetPort();
-  std::string const& host = request.GetUrl().GetScheme() + request.GetUrl().GetHost()
```

Comment:
How did this code ever work given that GetScheme() didn't append `://` to the scheme?

Reply:
Answering my own question: "host" was only ever used in output strings, except it is also used for the GetConnectionKey as the base for the key, and the key is only used when managing the connection pool. 

So the actual value of "host" is irrelevant, and thus adding the :// to the string doesn't matter.

IMHO, This is one of those cases where a brief comment ("Generate a display name for the host being connected to") or a somewhat more descriptive name (`hostDisplayName`) might make more sense - I had to spend 10 minutes reading the code trying to understand why this change was necesssary - my original thought was that the pull request changed how `GetScheme()` worked, and thus I tried to find out why this wasn't also a breaking change to the `GetScheme()` API.

Reply:
I'll move this to another PR that improves a Transport Exception output

Reply:
https://github.com/Azure/azure-sdk-for-cpp/pull/3456